### PR TITLE
[script][common-theurgy] First pass at sprinkle_holy_water?

### PR DESCRIPTION
**Pull request by MahtraDR**
First time making this kind of function... Does this work?

### CODE REVIEW

File: common-theurgy.lic [hunk 1]
```diff
@@ -142,12 +142,30 @@ module DRCTH
     DRCI.put_away_item?(DRC.left_hand, container)
   end
 
+  def sprinkle_holy_water?(theurgy_supply_container, water_holder, target)
+    did_sprinkle_holy_water = (
+      DRCI.get_item?(water_holder, theurgy_supply_container) &&
+      sprinkle?(water_holder, target) &&
+      DRCI.put_away_item?(water_holder, theurgy_supply_container)
+    (
```
Comment (KatoakDR) [inline common-theurgy.lic:150]: 🧹  need closing paren
```suggestion
    )
```

Comment (KatoakDR) [reply common-theurgy.lic:150]: This looks to be doing exactly the same as the existing `sprinkle_holy_water` method but just returns a `true` if all three actions are true, else `false`.

Is it necessary to have the new method or simply add the `&&` to the existing methods?

Comment (MahtraDR) [reply common-theurgy.lic]: I don't intend to keep both. I'll clean up when we have these, and remove the non-booleans.

File: common-theurgy.lic [hunk 2]
```diff
@@ -142,12 +142,30 @@ module DRCTH
     DRCI.put_away_item?(DRC.left_hand, container)
   end
 
+  def sprinkle_holy_water?(theurgy_supply_container, water_holder, target)
+    did_sprinkle_holy_water = (
```
Comment (rpherbig) [inline common-theurgy.lic:146]: Capturing the result and returning it isn't necessary unless you feel the var name adds expressiveness

The "Ruby way" would be
```suggestion
def sprinkle_holy_water?(theurgy_supply_container, water_holder, target)
      DRCI.get_item?(water_holder, theurgy_supply_container) && sprinkle?(water_holder, target) && DRCI.put_away_item?(water_holder, theurgy_supply_container)
end
```

(or add some newlines if you prefer that)

Ruby functions implicitly return the last thing in the function

Comment (MahtraDR) [reply common-theurgy.lic:146]: You and @asechrest said the same thing :) Great minds and all that. Let me swap that. I originally thought the variable name made things more explicit, but if the name of the function isn't explicit enough... that's a different problem.

Comment (MahtraDR) [reply common-theurgy.lic]: Like this?

File: common-theurgy.lic [hunk 3]
```diff
@@ -142,12 +142,24 @@ module DRCTH
...
   def sprinkle_holy_water(theurgy_supply_container, water_holder, target)
     DRCI.get_item?(water_holder, theurgy_supply_container)
     sprinkle?(water_holder, target)
     DRCI.put_away_item?(water_holder, theurgy_supply_container)
   end
 
+  def sprinkle_holy_oil?(theurgy_supply_container, target)
+      DRCI.get_item?('holy oil', theurgy_supply_container) &&
+      sprinkle?('oil', target) &&
+      DRCI.put_away_item?('holy oil', theurgy_supply_container) if DRCI.in_hands?('oil')
```
Comment (rpherbig) [inline common-theurgy.lic:160]: Sorry, I didn't see this the first time I looked at the diff.

I don't think I've seen an `if` in a chained conditional like this... Have you tested this helper to make sure it puts away oil if it's in hand and skips (but returns true!) if not?

Comment (MahtraDR) [reply common-theurgy.lic]: How's the current version? I've tried to copy the format from `wave_incense?`

**Comment by MahtraDR** [comment]
When we have this worked out, I'll make an oil version too.

File: common-theurgy.lic [hunk 4]
```diff
@@ -142,6 +142,20 @@ module DRCTH
     DRCI.put_away_item?(DRC.left_hand, container)
   end
 
+  def sprinkle_holy_water?(theurgy_supply_container, water_holder, target)
+    empty_cleric_hands(theurgy_supply_container)
```
Comment (rpherbig) [inline common-theurgy.lic:146]: Why empty hand first?

File: common-theurgy.lic [hunk 5]
```diff
@@ -142,6 +142,20 @@ module DRCTH
...
+    unless DRCI.get_item?(water_holder, theurgy_supply_container)
+      DRC.message("Can't get #{water_holder}, skipping Meraud commune.")
+      return false
+    end
+    unless sprinkle?(water_holder, target)
+      DRC.message("Couldn't sprinkle holy water, skipping Meraud commune.")
+      return false
```
Comment (rpherbig) [inline common-theurgy.lic:153]: Should we empty hands here?

File: common-theurgy.lic (hunk 4 above)
Comment (MahtraDR) [reply common-theurgy.lic]: I thought we needed to, but we don't. It works just the same on the left hand. Tested just now.

File: common-theurgy.lic (hunk 5 above)
Comment (MahtraDR) [reply common-theurgy.lic:153]: Added,

**Pull request merged**